### DISCUSSION TITLE
Pyleecan Object Attr. of Frozen Class

### DISCUSSION THREAD

**Discussion by SebGue** [Eomys/pyleecan#502] [Q&A] [answered]
Hello @BonneelP 

is it possible to have a object attribute of typ 'frozen' class? I never tried that but maybe you can tell me right away.
A use case could be a collection (list) of arbitrary pyleecan objects, e.g. a list of standardized stators. 
Just asking ... I don't need such feature but it may be interessting in the future. But then 'frozen' may need another name or another object 'layer' inbetween frozen and all other pyleecan objects.

Best regards, Sebastian

**Comment by BonneelP** [comment, marked as answer]
Hello Sebastian,

I haven't checked but it should work. We could introduce a "Pyleecan" type csv convention  and change the check type if needed to avoid writing "FrozenClass" in csv and not having the need to introduce a new class. 
We also have the possibility to not provide type at all but then it would require to work on the setter/getter to call the proper as_dict/to_list... It was necessary for DataKeeper or XOutput reference (I don't remember which one). 

Best regards,
Pierre
